### DISCUSSION TITLE
Feature request: Add a tps-catchup-max option

### DISCUSSION THREAD

**Discussion by PQCraft** [PurpurMC/Purpur#1306] [Feature Requests]
I have a potato "server" and `tps-catchup` has caused deaths by creepers before. I think adding a `tps-catchup-max` option would counter this a bit. For example, setting `tps-catchup-max` to `5` will only catch up at most 5 ticks per tick. I think another way this could be implemented is something like `tps-catchup-time-min` which could set the minimum milliseconds a catchup tick would take. For example, setting it to `40` would prevent catchup ticks from going too fast and limit the rate to ~25 ticks. You could probably do it where `tps-catchup-max` would calculate the minimum milliseconds for each catchup tick.

**Comment by granny** [comment]
The best counter to `tps-catchup` is disabling it outright. That's why the option exists, so that you can stop spikes of rushed ticks from happening.

**Reply by PQCraft** [reply]
Yeah but I would like it to catchup, just not to a ridiculous amount.
Like instead of making up all the ticks almost immediately, I would like it to make them up gradually.
Not catching up is bad and catching up too quickly is also bad.
